geode-cq/src/distributedTest/java/org/apache/geode/cache/query/dunit/QueryMonitorDUnitTest.java

```diff
@@ -416,24 +416,27 @@ private static void verifyException(Exception e) {
         + "\"Query execution taking more than the max execution time\"" + "\n Found \n" + error);
   }
 
-  private static class QueryTimeoutHook implements DefaultQuery.TestHook {
-    /**
-     * Delay query execution long enough for the QueryMonitor to mark the query as cancelled.
-     */
-    public void doTestHook(int spot, DefaultQuery query) {
-      if (spot != 6) {
-        return;
-      }
-
-      /*
-       * The pollDelay() value was chosen to be larger than the
-       * GemFireCacheImpl.MAX_QUERY_EXECUTION_TIME (system property) value,
-       * set during test initialization.
-       */
-      await("stall the query execution so that it gets cancelled")
-          .pollDelay(10, TimeUnit.MILLISECONDS)
-          .until(() -> query.isCanceled());
+  /**
+   * This method is used as an implementation of the TestHook functional interface, to delay
+   * query execution long enough for the QueryMonitor to mark the query as cancelled.
+   */
+  private static void delayQueryTestHook(final DefaultQuery.TestHook.SPOTS spot,
+      final DefaultQuery query) {
+    if (spot != DefaultQuery.TestHook.SPOTS.BEFORE_QUERY_DEPENDENCY_COMPUTATION) {
+      return;
     }
+    if (query.isCqQuery()) {
+      return;
+    }
+
+    /*
+     * The pollDelay() value was chosen to be larger than the
+     * GemFireCacheImpl.MAX_QUERY_EXECUTION_TIME (system property) value,
+     * set during test initialization.
+     */
+    GeodeAwaitility.await("stall the query execution so that it gets cancelled")
```

Comment:
Can we use a static import here instead?

Reply:
did it!